lib/signs_ui/signs/messages.ex

```diff
@@ -24,8 +24,22 @@ defmodule SignsUi.Signs.Messages do
   end
 
   def handle_call({:add_message, message}, _from, messages) do
-    SignsUiWeb.Endpoint.broadcast!("signs:all", "new_msg", %{body: "#{inspect(message)}"})
-    messages = Enum.take([message | messages], 10)
+    sta = message["sta"]
+    [duration, zone_line, text] = String.split(message["c"], ["~", "-"])
```

Comment:
I have some comments on parsing this info generally (e.g. maybe using a Regex, and allowing for the different components to be in a different order, which I think is allowed), but it's probably all superseded by the custom parser stuff you're looking into now related to Phoenix eating the duplicate command, so I think this is fine for our first pass.